src/ADAL.PCL.Desktop/WindowsFormsWebAuthenticationDialogBase.cs

```diff
@@ -181,13 +186,18 @@ private bool CheckForClosingUrl(Uri url)
         {
             // Make change here
             bool canClose = false;
-            if (url.Authority.Equals(this.desiredCallbackUri.Authority, StringComparison.OrdinalIgnoreCase) && url.AbsolutePath.Equals(this.desiredCallbackUri.AbsolutePath))
+            if (url.Authority.Equals(this.desiredCallbackUri.Authority, StringComparison.CurrentCultureIgnoreCase) &&
+                url.AbsolutePath.Equals(this.desiredCallbackUri.AbsolutePath, StringComparison.CurrentCulture))
             {
                 this.Result = new AuthorizationResult(AuthorizationStatus.Success, url.OriginalString);
                 canClose = true;
             }
-            
-            if (!canClose && !url.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase))
+
+            // if redirect_uri is not hit and scheme of the url is not whitelisted and url is not about:blank
+            // and url scheme is not https then fail to load.
+            if (!canClose && !whiteListedSchemes.Contains(url.Scheme.ToLower(CultureInfo.CurrentCulture)) &&
+                url.AbsoluteUri.Equals("about:blank", StringComparison.CurrentCultureIgnoreCase)
```

Comment:
shouldn't be !url.AbsoluteUri.Equals("about:blank", StringComparison.CurrentCultureIgnoreCase)